### PR TITLE
[fingerprint] Convert cli to typescript

### PR DESCRIPTION
**Pull request by wschurman**
# Why

I'd like to make EAS Build call the versioned CLI to do the fingerprint diff rather than duplicate the functions there. This has the benefit of being able to test changes to the diff/fingerprint in a build without requiring publishing and deploying a change to eas-build.

The first step is to convert this to typescript to make it easier to write safely as it is becoming more complex.

# How

Convert to typescript, put in cli module directory. Update package scripts.

# Test Plan

`et cp @expo/fingerprint`

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by wschurman** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/expo/expo/32487?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#32487** <a href="https://app.graphite.dev/github/pr/expo/expo/32487?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#32486** <a href="https://app.graphite.dev/github/pr/expo/expo/32486?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wschurman and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/@expo/fingerprint/src/cli.ts
Comment (Kudo) [inline packages/@expo/fingerprint/src/cli.ts:1]: empty file

**Review by Kudo — CHANGES_REQUESTED** [review]
please help to remove the empty file change and also update the publish files at https://github.com/expo/expo/blob/63b63942d42519e7914199010549cf08ff6c6b2a/packages/%40expo/fingerprint/package.json#L21-L24

**Review by Kudo — APPROVED** [review]

**Comment by Kudo** [comment]
maybe need to rebuild js files because check-packages is failed

**Comment by wschurman** [comment]
### Merge activity

* **Oct 31, 3:13 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/expo/32487).

**Pull request merged**